File: skills/tech-debt-scan/tests/fixtures/corpus/web-ts/files/src/report/export.ts
const KINDS = ["card", "bank", "wire"] as const;

export async function exportRows(rows: { url: string }[]): Promise<string[]> {
  const out: string[] = [];
  for (const row of rows) {
    const response = await fetch(row.url); // planted TD-36: I/O inside the loop
    out.push(await response.text());
  }
  return out;
}

export function kindLabels(): string[] {
  const labels: string[] = [];
  for (const kind of KINDS) { // decoy: a loop over a fixed three-member tuple has a bounded N
    labels.push(kind.toUpperCase());
  }
  return labels;
}

File: skills/tech-debt-scan/tests/fixtures/corpus/service-py/files/src/pay/export.py
"""Statement export."""
import re
from enum import Enum

HEADER_RE = re.compile(r"^#")  # compiled once at import: not a loop cost (decoy)


class Kind(Enum):
    CARD = "card"
    BANK = "bank"


def export_statements(paths):
    lines = []
    for path in paths:
        handle = open(path)  # planted TD-36: I/O inside the loop
        for line in handle:
            lines.append(line.strip())
    return lines


def kind_labels():
    labels = []
    for kind in Kind:  # decoy: a loop over an enum has a bounded N
        labels.append(kind.value.upper())
    return labels


def format_amount(cents):
    return f"{cents / 100:.2f}"
